src/gui/torrentcreatordlg.cpp

```diff
@@ -28,254 +28,350 @@
+    return pieceSizes[ui->comboPieceSize->currentIndex()] * 1024;
+}
+
+void TorrentCreatorDlg::dropEvent(QDropEvent *event)
+{
+    event->acceptProposedAction();
+
+    if (event->mimeData()->hasUrls()) {
+        // only take the first one
+        QUrl firstItem = event->mimeData()->urls().first();
+        QString path = (firstItem.scheme().compare("file", Qt::CaseInsensitive) == 0) ?
+                        firstItem.toLocalFile() : firstItem.toString();
+        ui->textInputPath->setText(path);
+    }
+}
+
+void TorrentCreatorDlg::dragEnterEvent(QDragEnterEvent *event)
+{
+    if (event->mimeData()->hasFormat("text/plain") || event->mimeData()->hasFormat("text/uri-list"))
+        event->acceptProposedAction();
 }
 
 // Main function that create a .torrent file
-void TorrentCreatorDlg::on_createButton_clicked()
+void TorrentCreatorDlg::onCreateButtonClicked()
 {
-    QString input = Utils::Fs::fromNativePath(textInputPath->text()).trimmed();
-    if (input.endsWith("/"))
-        input.chop(1);
-    if (input.isEmpty()) {
-        QMessageBox::critical(0, tr("No input path set"), tr("Please type an input path first"));
+    QString input = Utils::Fs::fromNativePath(ui->textInputPath->text()).trimmed();
+
+    // test if readable
+    QFileInfo fi(input);
+    if (!fi.isReadable()) {
+        QMessageBox::critical(this, tr("Torrent creator failed"), tr("Reason: Path to file/folder is not readable."));
         return;
     }
-    QStringList trackers = trackers_list->toPlainText().split("\n");
-    if (!trackers_list->toPlainText().trimmed().isEmpty())
-        saveTrackerList();
-
-    Preferences* const pref = Preferences::instance();
-    QString lastPath = pref->getCreateTorLastSavePath();
+    input = fi.canonicalFilePath();
 
-    QString destination = QFileDialog::getSaveFileName(this, tr("Select destination torrent file"), lastPath, tr("Torrent Files (*.torrent)"));
+    // get save path
+    QString lastPath = getLastSavePath();
+    QString destination = QFileDialog::getSaveFileName(this, tr("Select where to save the new torrent"), lastPath, tr("Torrent Files (*.torrent)"));
     if (destination.isEmpty())
         return;
+    if (!destination.endsWith(".torrent", Qt::CaseInsensitive))
+        destination += ".torrent";
+    setLastSavePath(Utils::Fs::branchPath(destination));
 
-    pref->setCreateTorLastSavePath(Utils::Fs::branchPath(destination));
-    if (!destination.toUpper().endsWith(".TORRENT"))
-        destination += QString::fromUtf8(".torrent");
-
-    // Disable dialog
+    // Disable dialog & set busy cursor
     setInteractionEnabled(false);
     showProgressBar(true);
-    // Set busy cursor
     setCursor(QCursor(Qt::WaitCursor));
-    // Actually create the torrent
-    QStringList urlSeeds = URLSeeds_list->toPlainText().split("\n");
-    QString comment = txt_comment->toPlainText();
+
+    QStringList trackers = ui->trackers_list->toPlainText().split("\n");
```

Comment:
It appear to do some tricks at: https://github.com/qbittorrent/qBittorrent/blob/master/src/base/bittorrent/torrentcreatorthread.cpp#L121